dashboard/src/pages/macros.tsx

```diff
@@ -42,66 +40,52 @@ const Macros = () => {
 
   const queryClient = useQueryClient();
 
-  const fetchMacros = async (instanceUuid: string) => {
-    const response: MacroEntry[] = await getMacros(instanceUuid);
-    setMacros(
-      response.map(
-        (macro, i) =>
-          ({
-            id: i + 1,
-            name: macro.name,
-            last_run: unixToFormattedTime(macro.last_run?.toString()),
-            path: macro.path,
-          } as TableRow)
-      )
-    );
-  };
-
-  const fetchTasks = async (instanceUuid: string) => {
-    const response = await getTasks(instanceUuid);
-    setTasks(
-      response.map(
-        (task, i) =>
-          ({
-            id: i + 1,
-            name: task.name,
-            creation_time: unixToFormattedTime(task.creation_time.toString()),
-            pid: task.pid,
-          } as TableRow)
-      )
-    );
-  };
-
-  const fetchHistory = async (instanceUuid: string) => {
-    const response = await getInstanceHistory(instanceUuid);
-    setHistory(
-      response.map(
-        (entry, i) =>
-          ({
-            id: i + 1,
-            name: entry.task.name,
-            creation_time: unixToFormattedTime(
-              entry.task.creation_time.toString()
-            ),
-            finished: unixToFormattedTime(entry.exit_status.time.toString()),
-            process_id: entry.task.pid,
-          } as TableRow)
-      )
-    );
-  };
-
-  useEffect(() => {
-    if (!selectedInstance) return;
+  const macros = useQuery(
+    ['instance', selectedInstance?.uuid, 'macroList'],
+    () => getMacros(selectedInstance?.uuid as string),
+    { enabled: !!selectedInstance, initialData: [], refetchOnMount: 'always' }
+  ).data.map(
+    (macro, i) =>
+      ({
+        id: i + 1,
+        name: macro.name,
+        last_run: unixToFormattedTime(macro.last_run?.toString()),
+        path: macro.path,
+      } as TableRow)
+  );
 
-    const fetchAll = async () => {
-      if (!selectedInstance) return;
-      fetchMacros(selectedInstance.uuid);
-      fetchTasks(selectedInstance.uuid);
-      fetchHistory(selectedInstance.uuid);
-    };
+  const tasks = useQuery(
+    ['instance', selectedInstance?.uuid, 'taskList'],
+    () => getTasks(selectedInstance?.uuid as string),
+    { enabled: !!selectedInstance, initialData: [], refetchOnMount: 'always' }
+  ).data.map(
+    (task, i) =>
+      ({
+        id: i + 1,
+        name: task.name,
+        creation_time: unixToFormattedTime(task.creation_time?.toString()),
+        pid: task.pid,
+      } as TableRow)
+  );
```

Comment:
It would be nice if we can unify our naming, (runningMacros vs taskList), also "tasks" and "tasklist" might be mistaken for one another.